### DISCUSSION TITLE
gpiozero multible leds

### DISCUSSION THREAD

**Discussion by mjs-pi** [gpiozero/gpiozero#1036] [Q&A]
Hi I have a python script to turn on leds on raspberry pi. I run to code below and and the leds all lght when accessing them 1 at a time but all leds do not light when i run the command leds.on()

from gpiozero import LEDBoard
from time import sleep

#leds = LEDBoard(5, 6, 13, 19, 26)
leds = LEDBoard(1,17,27,4,18,23, 24,25,22,5)

for led in leds:
    led.on()
    sleep(.2)
    led.off()

leds.on()
sleep(1)

**Comment by lurch** [comment]
Hmmm, from what I remember that code _should_ work.
Can you change the last two lines of your script to instead say:
```python
print("Turning off ledboard")
leds.off()
for led in leds:
    print(led.is_lit)
print("Turning on ledboard")
leds.on()
sleep(1)
for led in leds:
    print(led.is_lit)
```
Also, you might want to use `pause` at the end of your script rather than `sleep(1)` - see https://gpiozero.readthedocs.io/en/stable/faq.html#how-do-i-keep-my-script-running

**Reply by mjs-pi** [reply]
Thanks for the reply,
Interesting, so you think the pins need to turn off first. So I guess is_lit  will tell me what's going on.
Made the changes you suggested will load on Pi tomorrow and see.  Goal is to feed he LEDS from a pattern list.

**Reply by lurch** [reply]
> so you think the pins need to turn off first

No, I don't think that, I just added it as "debug code" in case the active-high / active-low was accidentally wired up or configured the wrong way.

> Goal is to feed he LEDS from a pattern list.

Yup, you should be able to set the `.value` of the entire LEDBoard with a tuple of boolean values, e.g. `leds.value = (True, True, False, True, False, False, False, True, False, False)`. And once you've got to grips with that you might find  https://gpiozero.readthedocs.io/en/stable/source_values.html useful too.

**Comment by mjs-pi** [comment]
Hi, 
Ran code with changes, weird?
Only pins 27,23,22 light up at end but
now,  even when i run the one at time, pins 1,17 light 
briefly then just pins 27,23,22 stay lit?

Is there something special about 27,23,22 pins?, looks like they should be just strait GPIO pins. 
As soon as i light one up, the rest go out?
confusing at best , i spend a lot of time tracking weird things down it seems :(. 

PS: If it matters I'm running this code on a PI model 2 B v1 from a while ago. If they ever get there pricing and stock under control,  I'm buying a newer one :)

leds = LEDBoard(1,17,27,4,18,23, 24,25,22,5)

Turning off ledboard
False, False, False, False, False, False, False, False, False, False, 
Turning on ledboard
True, True, True, True, True, True, True, True, True, True, 

but only pins 27,23,22 light up 

Thanks Again for your time

**Reply by lurch** [reply]
Sounds like either your wiring is a bit loose, or maybe you've accidentally damaged the GPIO pins on your Pi in the past, which is making them unreliable?
From a software point of view everything looks fine, so this sounds like a "hardware" problem.

The other possibility _might_ be that some background-software on your Pi is trying to control the GPIO pins at the same time, and the easiest way of ruling out that option would be to try flashing a fresh copy of Raspberry Pi OS onto a spare SD card.

**Comment by mjs-pi** [comment]
Thanks, Think wiring is OK, when I ran test again all 10 led light one at a time, but when trying to light all as long as pins 27,23,22, led 3,6,9 are not lit is OK. Attached a few pictures so you can see what I'm work on, Not sure I can do that?
![led_board_test](https://user-images.githubusercontent.com/101467474/158069257-836ba623-bbe6-4257-a431-40954f2d26e2.png)
.

**Reply by lurch** [reply]
Ah, looks like you've only got one resistor for all 10 LEDs? I think that'll be your problem. Different colour LEDs have different forward-voltages, and so if you're funnelling them all through a single resistor, that would explain why they all work when you switch them on one at a time, but don't all light up when you try to turn them all on at once.
Fix is to have a separate resistor for each LED.

**Comment by mjs-pi** [comment]
Sound reasonable,  I will rewire with at least separate resistors on the red one first,  they seem to be the trouble makers.
Once again thanks. I will try not to bother you again if i can. you have been great. Mike

**Reply by lurch** [reply]
No problem at all, I'm always happy to help :+1: 
Looks like https://www.google.com/search?q=can+i+use+one+resistor+for+multiple+leds has lots of useful info.

**Comment by mjs-pi** [comment]
Man you know your stuff, looks like lots of interesting reading here,  everyday is a learning experience and that is the way it should be. Just glad I'm retired and have time to tinker  as my wife calls it : )
Thanks Mike

**Comment by mjs-pi** [comment]
Hi
 Thanks for all your help, separate resistors  on the red led fix it. Now work as intend. Have patterns working as well.
Like i said your the man.  Now i have some reading to do about using resistors on led .
Mike

**Reply by lurch** [reply]
The other benefit to using one-resistor-per-LED is that each LED will have the same brightness regardless of how many LEDs you have switched on. Whereas if you have multiple LEDs all going through a single resistor, each LED will get dimmer the more LEDs you switch on.
